### PR TITLE
Add macFUSE check for MacOS

### PR DESCRIPTION
**Pull request by Haoning-Sun**
### What changes are proposed in this pull request?

Add macFUSE check for MacOS.

### Why are the changes needed?
The method isFuseInstalled can not check macFUSE.

### Does this PR introduce any user facing changes?
no.

### CODE REVIEW

File: integration/fuse/src/main/java/alluxio/fuse/AlluxioFuseUtils.java
Comment (LuQQiu) [inline integration/fuse/src/main/java/alluxio/fuse/AlluxioFuseUtils.java:382]: Are there existing java function or Alluxio java function do the same thing? so that we don't need to create our own

Comment (Haoning-Sun) [reply integration/fuse/src/main/java/alluxio/fuse/AlluxioFuseUtils.java]: I didn't find the relevant method in alluxio, ready to use ComparableVersion to replace the custom method.

File: pom.xml
Comment (Xenorith) [inline pom.xml:697]: if we add new dependencies, please specify the license governing the usage of this license to ensure it respects our license model. in most cases, these common dependencies are apache 2.0, but we want to confirm before merging

File: integration/fuse/src/main/java/alluxio/fuse/AlluxioFuseUtils.java
Comment (LuQQiu) [inline integration/fuse/src/main/java/alluxio/fuse/AlluxioFuseUtils.java:372]: if `comparable version` requires a new dependency, then may be worth using the original approach.

Comment (LuQQiu) [reply integration/fuse/src/main/java/alluxio/fuse/AlluxioFuseUtils.java:372]: sorry for the extra work introduced. I thought probably in the code base there always some version comparison methods included

Comment (Haoning-Sun) [reply integration/fuse/src/main/java/alluxio/fuse/AlluxioFuseUtils.java]: I reverted to the original method.

File: integration/fuse/src/main/java/alluxio/fuse/AlluxioFuseUtils.java
Comment (yyongycy) [inline integration/fuse/src/main/java/alluxio/fuse/AlluxioFuseUtils.java:370]: this just means the version is supported by alluxio, not actually means the mac or not?

Comment (Haoning-Sun) [reply integration/fuse/src/main/java/alluxio/fuse/AlluxioFuseUtils.java:370]: This has nothing to do with alluxio, but to determine the version of fuse supported by different versions of mac os.

**Review by yyongycy — APPROVED** [review]
LGTM as long as you tested this change.

**Comment by Haoning-Sun** [comment]
@yyongycy MacFuse version has been tested.

**Comment by yyongycy** [comment]
@LuQQiu any more comments?

**Comment by Haoning-Sun** [comment]
@LuQQiu Please help review this pr, thank you!

**Review by LuQQiu — APPROVED** [review]

**Comment by LuQQiu** [comment]
alluxio-bot, merge this please

**Pull request merged**
